### PR TITLE
FeatureToggles: Add context and and an explicit global check

### PR DESCRIPTION
**Pull request by ryantxu**
Currently we use feature flags to toggle an enormous range of behavior.  It may be simple frontend only changes, or swap our entire startup sequences.

As we move towards cloud first deployments and allowing people to easily change the settings, we need better mechanisms to distinguish between the global changes and ones that could be considered as user options.

This PR updates our interface so we now have two functions: one with context and one for global.
image

everywhere a context is easily available, I added it; otherwise it gets the global function.

Over the next few months, we should move to enforce that features requested through the global checks are actually marked as global in the registry and do not allow end-user configuration.

### CODE REVIEW

**Comment by ryantxu** [comment]
@stephaniehingtgen -- looks like this task does not try to checkout the enterprise branch with the same name 🤔  https://github.com/grafana/grafana-enterprise/pull/6031

File: pkg/services/featuremgmt/models.go [hunk 1]
```diff
@@ -2,11 +2,21 @@ package featuremgmt
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
 )
 
 type FeatureToggles interface {
-	IsEnabled(flag string) bool
+	// Check if a feature is enabled for a given request.
```
Comment (bergquist) [inline pkg/services/featuremgmt/models.go:10]:
```suggestion
	// Check if a feature is enabled for a given context.
```

background jobs etc.

**Review by bergquist — APPROVED** [review]
Love this! But please make it start in a ephemeral instance first :)

**Pull request merged**